stablehlo/dialect/TypeInference.cpp

```diff
@@ -1141,8 +1136,14 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
       return emitOptionalError(location, "stride must be positive but got ",
                                strideVals[i], " in dimension ", i);
 
-    shape.push_back(inferSliceDim(rankedTy.getDimSize(i), start[i], limit[i],
-                                  strideVals[i]));
+    if (!hlo::isDynamicDimSize(rankedTy.getDimSize(i))) {  // static dim
+      shape.push_back(inferSliceDim(rankedTy.getDimSize(i), start[i], limit[i],
+                                    strideVals[i]));
+      if (!inputBounds.empty())
+        resultBounds.push_back(ShapedType::kDynamicSize);
+    } else  // dynamic dim with bounds
+      resultBounds.push_back(
```

Comment:
let's add a test demonstrating the issue this PR is fixing?

Reply:
Done.